ldap/src/main/java/org/springframework/security/ldap/DefaultSpringSecurityContextSource.java

```diff
@@ -171,4 +173,14 @@ private static String buildProviderUrl(List<String> urls, String baseDn) {
 
 	}
 
+	/**
+	 * Create and initialize an instance which will connect to the embedded LDAP server.
+	 *
+	 * @param container the embedded DS container
+	 * @return an instance which will connect to the embedded LDAP server
+	 */
+	public static DefaultSpringSecurityContextSource createEmbeddedContextSource(EmbeddedDsContainer container) {
```

Comment:
I don't think we should need to add this method. I'd prefer to keep as many things private scope as possible.

Reply:
I added this factory method for XML config because we need to create and post process a bean of embedded container first and then use actual port value to build a url.

Reply:
This can likely be done with a package private method within the config itself.

Reply:
Could you please provide some example of how this can be done in BeanDefinitionParser?

Reply:
Got it. Added `EmbeddedLdapServerConfigBean` with factory method.